cohttp-eio/src/cohttp_eio.mli

```diff
@@ -93,10 +93,20 @@ module Server : sig
 
   val run :
     ?socket_backlog:int -> ?domains:int -> port:int -> 'a env -> handler -> 'b
+  (** [run ~socket_backlog ~domains ~port env handler] runs a HTTP/1.1 server
+      executing [handler] and listening on [port]. [env] corresponds to
+      {!val:Eio.Stdenv.t}.
+
+      [socket_backlog] is the number of pending connections for tcp server
+      socket. The default is [128].
+
+      [domains] is the number of OCaml 5.0 domains the server will use. The
```

Comment:
I still think this is a little ambiguous (I did like the idea of having two functions to be very explicit about this)?

 - Does having `domains:1` mean I'm spawning one extra domain or is it single-threaded? I know the answer is the latter but I still think people might be confused?
 - If we go with this API I think it has to be stated that `handler` should be a domain-safe function

Reply:
> Does having domains:1 mean I'm spawning one extra domain or is it single-threaded? I know the answer is the latter but I still think people might be confused?

This should be addressed (multiple specific functions and docs on `handler` being thread-safe) in future PR along with a few other suggestions from @talex5.